lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -709,6 +709,10 @@ LogicalResult NEQPrimOp::canonicalize(NEQPrimOp op, PatternRewriter &rewriter) {
 //===----------------------------------------------------------------------===//
 
 OpFoldResult AsSIntPrimOp::fold(ArrayRef<Attribute> operands) {
+  // No effect.
+  if (input().getType() == getType())
+      return input();
+
```

Comment:
These are neat! Reset inference has a tendency to make casts on resets no-ops. This should neatly clean them up.